crates/primitives/src/bridge.rs

```diff
@@ -287,22 +319,42 @@ impl From<SecNonce> for Musig2SecNonce {
 }
 
 impl BorshSerialize for Musig2SecNonce {
-    fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
-        writer.write_all(&self.0.to_bytes())
+    fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        let nonce_bytes = self.0.serialize();
+
+        // Write the length prefix as a u32 (little-endian)
+        borsh::BorshSerialize::serialize(&(nonce_bytes.len() as u32), writer)?;
+
+        // Write the nonce bytes
+        writer.write_all(&nonce_bytes)
     }
 }
 
 impl BorshDeserialize for Musig2SecNonce {
-    fn deserialize_reader<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
-        let mut sec_nonce_bytes = [0u8; 64];
-        reader.read_exact(&mut sec_nonce_bytes)?;
+    fn deserialize_reader<R: Read>(reader: &mut R) -> std::io::Result<Self> {
+        // Read the length prefix as a u32 (little-endian)
+        let len = u32::deserialize_reader(reader)? as usize;
+
+        if len != SEC_NONCE_SIZE {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                format!(
+                    "Invalid Musig2SecNonce size, expected: {}, got: {}",
+                    PUB_NONCE_SIZE, len
+                ),
+            ));
```

Comment:
And here.

Reply:
Done in https://github.com/alpenlabs/strata/commit/3bfcb5b8058f663b2fd7416ca9660e5d2060aefb.